plugins/deprecation.nix

```
@@ -36,5 +51,36 @@ in
           "plugins"
           new
         ]
-    ) renamed);
+    ) renamed)
+    ++ builtins.map (
+      name:
+      lib.mkRemovedOptionModule
+        [
+          "plugins"
+          name
+          "iconsPackage"
+        ]
+        ''
+          Please use `plugins.web-devicons` or `plugins.mini.modules.icons` with `plugins.mini.mockDevIcons` instead.
+        ''
+    ) iconsPackagePlugins
+    # Show a warning when web-devicons is auto-enabled
+    ++ [
+      (
+        { config, options, ... }:
+        {
+          config = lib.mkIf (options.plugins.web-devicons.enable.highestPrio == 1490) {
+            warnings = [
+              ''
+                Nixvim: `plugins.web-devicons` was enabled automatically because one of the following plugins was enabled.
```

```suggestion
                Nixvim: `plugins.web-devicons` was enabled automatically because the following plugins are enabled.
```

(My mistake) because we're filtering the list below. Alternatively we could keep the "one of" wording and remove the filter.